src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -216,6 +219,7 @@ create_distributed_table(PG_FUNCTION_ARGS)
 	 * backends manipulating this relation.
 	 */
 	Relation relation = try_relation_open(relationId, ExclusiveLock);
+	relation_close(relation, NoLock);
```

Comment:
could directly use LockRelation or something, or should comment why we close relation immediately

Reply:
```C
	/*
	 * Lock target relation with an exclusive lock - there's no way to make
	 * sense of this table until we've committed, and we don't want multiple
	 * backends manipulating this relation.
	 */
	Relation relation = try_relation_open(relationId, ExclusiveLock);

	if (relation == NULL)
	{
		ereport(ERROR, (errmsg("could not create distributed table: "
							   "relation does not exist")));
	}
	relation_close(relation, NoLock);

```

could be something like above

Reply:
now that we use `try_relation_open` and error out if relation does not exist, resolving my comment

Reply:
48db49e08018d81cc899f0106b4eb49f28beebcf